File: Algorithms/p275_H-Index_II.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        n = len(citations)
        l =0
        r = n
        while l<r:
            mid = (l+r)//2
            if citations[mid]<n-mid:
                l = mid+1
            elif citations[mid]>n-mid:
                r = mid
            else:
                return n-mid
        return n-l

File: Algorithms/p275_H-Index_II/p275_H-Index_II.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int n = citations.size();
        int l =0;
        int r = n;
        while(l<r){
            int mid = (l+r)/2;
            if(citations[mid]<n-mid){
                l = mid+1;
            }else if(citations[mid]>n-mid){
                r = mid;
            }else{
                return n-mid;
            }
        }
        return n-l;
    }
};
